### PR TITLE
[dev-launcher][ios] Adjust config plugin for new bare template - SDK 43

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Closes ENG-2077.

# How

Apply changes describe here: https://github.com/expo/expo/pull/14617.

# Test Plan

- https://exponent-internal.slack.com/archives/C011V63429H/p1633039856117100 ✅
- added tests ✅

### CODE REVIEW

File: packages/expo-dev-launcher/plugin/src/withDevLauncher.ts [hunk 1]
```diff
@@ -220,7 +195,7 @@ const withDevLauncherPodfile: ConfigPlugin = (config) => {
     'ios',
     async (config) => {
       await editPodfile(config, (podfile) => {
-        podfile = podfile.replace("platform :ios, '10.0'", "platform :ios, '11.0'");
+        podfile = podfile.replace("platform :ios, '10.0'", "platform :ios, '12.0'");
```
Comment (esamelson) [inline packages/expo-dev-launcher/plugin/src/withDevLauncher.ts:198]: is there any need to also check for `platform :ios, '11.0'`?

File: packages/expo-dev-launcher/plugin/src/withDevLauncherAppDelegate.ts [hunk 2]
```diff
@@ -106,30 +207,61 @@ export function modifyAppDelegate(appDelegate: string, expoUpdatesVersion: strin
...
-  if (!appDelegate.includes(DEV_LAUNCHER_APP_DELEGATE_ON_DEEP_LINK)) {
+  appDelegate = removeDevMenuInit(appDelegate);
+  return appDelegate;
+}
+
+export function modifyAppDelegate(appDelegate: string, expoUpdatesVersion: string | null = null) {
+  const shouldAddUpdatesIntegration =
+    expoUpdatesVersion != null && semver.gt(expoUpdatesVersion, '0.6.0');
+
+  if (!appDelegate.includes(DEV_LAUNCHER_INITIALIZE_REACT_NATIVE_APP_FUNCTION_DEFINITION)) {
```
Comment (esamelson) [inline packages/expo-dev-launcher/plugin/src/withDevLauncherAppDelegate.ts:225]: this seems like a really brittle check -- any reason we couldn't use `NEW_INITIALIZE_REACT_NATIVE_APP_FUNCTION` instead?

File: packages/expo-dev-launcher/plugin/src/withDevLauncherAppDelegate.ts [hunk 3]
```diff
@@ -62,10 +81,51 @@ const DEV_MENU_IOS_INIT = `
   [DevMenuManager configureWithBridge:bridge];
 #endif`;
 
-export function modifyAppDelegate(appDelegate: string, expoUpdatesVersion: string | null = null) {
-  const shouldAddUpdatesIntegration =
-    expoUpdatesVersion != null && semver.gt(expoUpdatesVersion, '0.6.0');
+const DEV_LAUNCHER_INIT_TO_REMOVE = `RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];
+  RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge moduleName:@"main" initialProperties:nil];
```
Comment (esamelson) [inline packages/expo-dev-launcher/plugin/src/withDevLauncherAppDelegate.ts:85]: Is there a way to do this without hardcoding the moduleName `main` here and on line 111?

Comment (lukmccall) [reply packages/expo-dev-launcher/plugin/src/withDevLauncherAppDelegate.ts:85]: If you want to change it, you can always provide the init function in AppDelegate. For our purpose,  the main should be enough.

**Pull request merged**